roles/validate_inventory/tasks/ai.yml

```diff
@@ -28,3 +28,36 @@
     fail_msg: "{{ item }} is not within the machine network!"
   when: vip_dhcp_allocation == false
   loop: "{{ groups['masters'] + (groups['workers'] | default([])) }}"  # This should not include day2_workers as they can be RWNs
+
+- name: Validate extra VIPs for dualstack
+  block:
+    - name: Assert that Openshift version is supported for dualstack VIPs
+      assert:
+        that:
+          - openshift_full_version is version('4.12', '>=')
+        fail_msg: "openshift_full_version  must be >= 4.12. to support dualstack VIPs"
+
+    - name: Assert extra api VIP is within the extra machine networks
+      assert:
+        that:
+          - hostvars['assisted_installer']['extra_api_vip'] | ansible.utils.ipaddr(item.cidr) | ansible.utils.ipaddr('bool')
```

Comment:
I don't think this is quite right. This is checking if the `api_vip` is within all the networks where as it should be in any of the extra networks. So I think you need to collect the tests into a list and then check the list with `is any` so that you don't false negatives.

Reply:
Thanks for the clarification, I think I have a better idea now, we should only fail the validation if none an extra vip is not in any of the `extra_machine_networks`,  if it matches one, it should pass right? 

I updated the PR, and I included some tests of the conditions [here](https://paste.openstack.org/show/bnblDSR3dt72xO80UvTT/)